### PR TITLE
Executed execution state changes finality fix

### PR DESCRIPTION
**Pull request by mateusz-sekara**
## Motivation

Fixing regression introduced here https://github.com/smartcontractkit/ccip/pull/353/files#diff-e4ddd08cbd5493f0d64d1f13004de2b93400ebe902b34515650f8ca17fe056a2L434

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go [hunk 1]
```diff
@@ -450,7 +451,7 @@ func (o *OffRamp) GetExecutionStateChangesBetweenSeqNums(ctx context.Context, se
 
 			return &cciptypes.ExecutionStateChanged{
 				SequenceNumber: sc.SequenceNumber,
-				Finalized:      sc.Raw.BlockNumber >= uint64(latestBlock.FinalizedBlockNumber),
+				Finalized:      sc.Raw.BlockNumber <= uint64(latestBlock.FinalizedBlockNumber),
```
Comment (makramkd) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go:454]: Do we have a test for this?

File: core/services/ocr2/plugins/ccip/ccipexec/ocr2.go [hunk 2]
```diff
@@ -1006,7 +1006,7 @@ func (r *ExecutionReportingPlugin) getUnexpiredCommitReports(
 	notSnoozedReports := make([]cciptypes.CommitStoreReport, 0)
 	for _, report := range reports {
 		if r.snoozedRoots.IsSnoozed(report.MerkleRoot) {
-			lggr.Debug("Skipping snoozed root", "minSeqNr", report.Interval.Min, "maxSeqNr", report.Interval.Max)
+			lggr.Debugw("Skipping snoozed root", "minSeqNr", report.Interval.Min, "maxSeqNr", report.Interval.Max)
```
Comment (roman-kashitsyn) [inline core/services/ocr2/plugins/ccip/ccipexec/ocr2.go]:
```suggestion
			lggr.Debugw("Skipping snoozed root", "minSeqNr", report.Interval.Min, "maxSeqNr", report.Interval.Max, "root", hex.EncodeToString(report.MerkleRoot[:]))
```

**Review by roman-kashitsyn — APPROVED** [review]
Great catch!
Inverting the comparison direction is classic; I made similar mistakes many times myself.

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go (hunk 1 above)
Comment (mateusz-sekara) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go:454]: Before the faulty PR https://github.com/smartcontractkit/ccip/pull/353/files#diff-e4ddd08cbd5493f0d64d1f13004de2b93400ebe902b34515650f8ca17fe056a2L434

```go
finalized := (latestBlock - stateChange.BlockNumber) >= int64(r.offchainConfig.DestFinalityDepth)
```

If you transform that you get
```go
finalized := latestBlock - int64(r.offchainConfig.DestFinalityDepth) >= state.BlockNumber
// and in this case 
latestFinalizedBlock := latestBlock - int64(r.offchainConfig.DestFinalityDepth)
```

Comment (mateusz-sekara) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go:454]: I'm gonna add tests as a separate PR

**Review by makramkd — APPROVED** [review]

**Pull request merged**